Migrate repo to ODK 1.5

matentzn opened the pull request.

matentzn commented on src/ontology/cob-edit.owl:
```diff
@@ -56,12 +56,6 @@
 
-    <!-- http://purl.org/dc/elements/1.1/date -->
-
-    <owl:AnnotationProperty rdf:about="http://purl.org/dc/elements/1.1/date"/>
```
Changed the dc elements namespace to dc terms throughout to satisfy a new, default QC check in ODK 1.5.

matentzn commented on src/ontology/imports/ro_import.owl:
Updated ro import according to new spec #260

matentzn commented on src/ontology/cob.Makefile:
```diff
@@ -103,6 +103,12 @@ $(COB_ANNOTATIONS): $(TMPDIR)/cob-annotations.ttl
+$(IMPORTDIR)/ro_import.owl: $(MIRRORDIR)/ro.owl $(IMPORTDIR)/ro_terms_combined.txt
+	if [ $(IMP) = true ]; then $(ROBOT) query -i $< --update ../sparql/preprocess-module.ru \
+		extract -T $(IMPORTDIR)/ro_terms_combined.txt --copy-ontology-annotations true --force true --individuals exclude --method BOT \
+		remove --base-iri http://purl.obolibrary.org/obo/COB_ --axioms internal --preserve-structure false --trim false \
```
This is the custom row: removing all COB related axioms from RO, but otherwise pulling in logical dependencies.

jamesaoverton reviewed (changes requested):
I think this is mostly fine but I'm worried about `cob-base.owl`.

(I'm not sure how valuable my review is. I don't use ODK much, and there's just a ton of changes here.)

This is what I did:

1. make a fresh clone
2. switch to this branch
3. `cd src/ontology/`
4. `./run.sh make all`

That mostly works, but I get an error at https://github.com/OBOFoundry/COB/blob/odk1.5/src/ontology/cob.Makefile#L26 because `products/*` does not match any files. If I `mkdir products && touch products/foo.bar` then `make all` completes successfully.

Then I look at the diffs in the generated files:

- `cob.owl`, `cob-full.owl` and `cob-native.owl` have minor differences that make sense -- good
- `src/ontology/imports/omo_import.owl` has a lot of additions, which I guess is fine
- `src/ontology/imports/ro_import.owl` has minor changes that I understand -- good
- `cob-base.owl` I'm worried about
    - drops a lot of assertions about external terms, which is good
    - also drops important equivalence axioms and annotations on them, which I don't think is good but I'm not sure; e.g. COB:0000107 equivalent to OBI:0000070
    - drops assertions entirely about COB, e.g. COB:0000016's inverse is COB:0000086

matentzn commented:
@jamesaoverton for the review.

> (I'm not sure how valuable my review is. I don't use ODK much, and there's just a ton of changes here.)

Very valuable. Thanks for taking the time!



> That mostly works, but I get an error at https://github.com/OBOFoundry/COB/blob/odk1.5/src/ontology/cob.Makefile#L26 because products/* does not match any files. If I mkdir products && touch products/foo.bar then make all completes successfully.

> I added a README.md now in there as a fixture so this should not happen again.


> cob-base.owl I'm worried about
> drops a lot of assertions about external terms, which is good
> also drops important equivalence axioms and annotations on them, which I don't think is good but I'm not sure; e.g. COB:0000107 equivalent to OBI:0000070

That should not be the case.. See below, I have reverted the process for now!

> drops assertions entirely about COB, e.g. COB:0000016's inverse is COB:0000086

Good spot!
This is because of some missing conditionals, which we will have in the next ROBOT release: https://github.com/ontodev/robot/pull/1187/files#diff-dc25d0b1a6794b1ac6ad2d5dfa891e182bc17f72541e3ca467cb3fd7a62a1e3eR1543

I have decided to revert the COB base process to what it was before. The accompanying issue is #261. Thanks for spotting that. COB base is not that straight forward to define I guess.

matentzn commented:
I also made sure that you and your team can run the releases again properly, by upgrading sssom py during the relevant pipeline step. I think this is ready to go now!

jamesaoverton commented:
Is this superseded by #275, and maybe #279?

matentzn commented:
Yes, I forgot this existed and scrambled to migrate your feedback over to `main` now, #279
